File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let result = 1;
    let clothesTable = new Map();
    clothes.map((c)=>{
        if(clothesTable.get(c[1])===undefined) clothesTable.set(c[1],1);
        else clothesTable.set(c[1],clothesTable.get(c[1])+1);
    });
    clothesTable.forEach((c)=>{
        result*=c+1;
    })
    console.log(result);
    return result-1;
}
// 경우의 수는 모든 옷 종류 + 옷 조합
// 옷 조합 = 모든 옷 곱하기

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>
#include <set>
using namespace std;
set<string> s;
map<string,vector<string>> m; 
int solution(vector<vector<string>> clothes) {
    int answer = 1;
    for(vector<string> ss : clothes){
        s.insert(ss[1]);
        m[ss[1]].push_back(ss[0]);
    }
    for(pair<string, vector<string>> mm: m){
        answer*= mm.second.size()+1;
    }
    return answer-1;
}